apex/contrib/optimizers/distributed_fused_lamb.py

```diff
@@ -647,7 +650,23 @@ def _reduce_scatter_and_all_reduce_scale(self, block_id, scale):
             rs_stream.wait_stream(torch.cuda.current_stream())
             rs_stream.wait_stream(self._l2_grad_norm_st)
             with torch.cuda.stream(rs_stream):
-                works[chunk_id] = torch.distributed.reduce_scatter(self._fp16_g_chunks[block_id][chunk_id],self._flat_grads_shards[block_id][chunk_id],group=self._rs_pg[glob_chunk_id%self._num_rs_pg],async_op=True,no_copy=True,op=_make_nccl_premul_sum((scale,)))
+                if self.no_copy_supported:
+                    works[chunk_id] = torch.distributed.reduce_scatter(
+                        output = self._fp16_g_chunks[block_id][chunk_id],
+                        input_list = self._flat_grads_shards[block_id][chunk_id],
+                        group = self._rs_pg[glob_chunk_id%self._num_rs_pg],
+                        async_op = True,
+                        no_copy = True,
+                        op = _make_nccl_premul_sum((scale,)),
+                    )
+                else:
+                    works[chunk_id] = torch.distributed._reduce_scatter_base(
```

Comment:
Perhaps we should use here `reduce_scatter_tensor` instead of `_reduce_scatter_base`. Will it be supported in 22.11?

Reply:
22.11 should have. we might need to check the availability of `reduce_scatter_tensor` (and `all_gather_into_tensor`) as well as no_copy check... 🤯